### PR TITLE
Add test for negative api ping

### PR DESCRIPTION
**Pull request by Gauravtalreja1**
**Description:**
- Add negative API ping to test to cover BZ-1835122.
- Move API ping tests from endtoend to api/test_ping.py to keep alike CLI tests.
- Update pytest marks for cli/test_ping.py.

**Test Results:**
```
$ pytest -qv --disable-warnings tests/foreman/api/test_ping.py 
=============================================================== test session starts ===============================================================
platform linux -- Python 3.8.2, pytest-6.2.5, py-1.10.0, pluggy-1.0.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/gtalreja/sat_workspace/robottelo, configfile: pyproject.toml
plugins: xdist-2.4.0, services-2.2.1, forked-1.3.0, reportportal-5.0.8, mock-3.6.1, ibutsu-1.16, cov-3.0.0
collected 2 items                                                                                                                                 

tests/foreman/api/test_ping.py ..                                                                                                           [100%]

========================================================= 2 passed, 4 warnings in 53.10s ==========================================================
```
```
$ pytest -qv --disable-warnings tests/foreman/cli/test_ping.py 
=============================================================== test session starts ===============================================================
platform linux -- Python 3.8.2, pytest-6.2.5, py-1.10.0, pluggy-1.0.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/gtalreja/sat_workspace/robottelo, configfile: pyproject.toml
plugins: xdist-2.4.0, services-2.2.1, forked-1.3.0, reportportal-5.0.8, mock-3.6.1, ibutsu-1.16, cov-3.0.0
collected 2 items                                                                                                                                 

tests/foreman/cli/test_ping.py ..                                                                                                           [100%]

========================================================= 2 passed, 3 warnings in 52.60s ==========================================================
```
Signed-off-by: Gaurav Talreja <gtalreja@redhat.com>

### CODE REVIEW

File: tests/foreman/api/test_ping.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    #    'candlepin_auth': {'duration_ms': '41', 'status': 'ok'},
+    #    …
+    # }, 'status': 'ok'}
+    services = response['services']
+    assert all(
+        [service['status'] == 'ok' for service in services.values()]
+    ), 'Not all services seem to be up and running!'
+
+
+def test_negative_ping_fail_status(default_sat):
```
Comment (pondrejk) [inline tests/foreman/api/test_ping.py:48]: we should mark this test as destructive as it stops a service

File: tests/foreman/api/test_ping.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    :customerscenario: true
+
+    :BZ: 1835122
+
+    :expectedresults: ping fails and shows FAIL status.
+    """
+    command_out = default_sat.execute('satellite-maintain service stop --only tomcat.service')
+    assert command_out.status == 0
+    response = default_sat.api.Ping().search_json()
+    assert response['status'] == 'FAIL'
```
Comment (pondrejk) [inline tests/foreman/api/test_ping.py:62]: should we assert more specifically on which services failed in addition to the overall status

File: tests/foreman/api/test_ping.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    """Negative test to verify FAIL status when ping fail
+
+    :id: 6bd2c552-29d7-4f49-81dd-f8d5ff0e7c39
+
+    :customerscenario: true
+
+    :BZ: 1835122
+
+    :expectedresults: ping fails and shows FAIL status.
+    """
```
Comment (pondrejk) [inline tests/foreman/api/test_ping.py:58]: Id say we should check all services are up when we start testing

**Review by pondrejk — CHANGES_REQUESTED** [review]

**Comment by devendra104** [comment]
@Gauravtalreja1 Converted the PR to draft state, Change the state once it gets ready for review.

**Comment by latran** [comment]
@Gauravtalreja1 any updates on this?

**Comment by JacobCallahan** [comment]
bump

**Comment by Gauravtalreja1** [comment]
Mistakenly deleted this branch locally, will restore and have plenty of things in todo, will continue this soon

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/destructive/test_ping.py

**Review by swadeley — APPROVED** [review]
ACK

File: tests/foreman/destructive/test_ping.py [hunk 4]
```diff
@@ -21,23 +21,44 @@
...
-    """Negative test to verify non-zero status code of ping fail
+@pytest.fixture(scope='module')
+def tomcat_service_teardown(request, module_target_sat):
+    assert module_target_sat.cli.Service.stop(options={'only': 'tomcat.service'}).status == 0
+
+    @request.addfinalizer
+    def _finalize():
+        assert module_target_sat.cli.Service.start(options={'only': 'tomcat.service'}).status == 0
+
+    yield module_target_sat
```
Comment (omkarkhatavkar) [inline tests/foreman/destructive/test_ping.py:32]: no need to yield satellite here, you can just yield and use `module_target_sat` in the test function.

File: tests/foreman/destructive/test_ping.py [hunk 5]
```diff
@@ -21,23 +21,44 @@
...
 
-    :CaseImportance: Critical
-
     :expectedresults: Hammer ping fails and returns non-zero(1) status code.
-
     """
-    command_out = target_sat.execute('satellite-maintain service stop --only tomcat.service')
-    assert command_out.status == 0
-    result = target_sat.execute("hammer ping")
+    result = tomcat_service_teardown.execute("hammer ping")
```
[the remark this answers is not in the document]
Comment (Gauravtalreja1) [reply tests/foreman/destructive/test_ping.py:46]: I kept this name by referring to the existing sys fixture `foreman_service_teardown`, closing this thread as non-blocking

**Review by omkarkhatavkar — APPROVED** [review]
